osu.Game/Overlays/MusicController.cs

```diff
@@ -419,6 +482,45 @@ public void ResetTrackAdjustments()
         }
     }
 
+    public class DuckParameters
+    {
+        /// <summary>
+        /// The duration of the ducking transition in milliseconds.
+        /// Defaults to 100 ms.
+        /// </summary>
+        public double DuckDuration = 100;
+
+        /// <summary>
+        /// The final volume which should be reached during ducking, when 0 is silent and 1 is original volume.
+        /// Defaults to 25%.
+        /// </summary>
+        public float DuckVolumeTo = 0.25f;
```

Comment:
Volume fields generally use `double` type instead of `float`, I believe mixing both together can result in stupid precision errors if I still remember how floating-type mathematics work.